src/Domain.LinnApps/ImportBooks/ImportBookService.cs

```diff
@@ -3,19 +3,84 @@
 
+        private readonly IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> orderDetailRepository;
+
+        private readonly IRepository<PurchaseLedger, int> purchaseLedgerRepository;
+
+
+        private readonly IPurchaseLedgerPack purchaseLedgerPack;
+
+        private readonly IQueryRepository<Supplier> supplierRepository;
+
+        private readonly ITransactionManager transactionManager;
+
         public ImportBookService(
             IRepository<ImportBookExchangeRate, ImportBookExchangeRateKey> exchangeRateRepository,
-            IRepository<LedgerPeriod, int> ledgerPeriodRepository)
+            IRepository<LedgerPeriod, int> ledgerPeriodRepository,
+            IQueryRepository<Supplier> supplierRepository,
+            IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> orderDetailRepository,
+            IRepository<PurchaseLedger, int> purchaseLedgerRepository,
+            ITransactionManager transactionManager,
+            IPurchaseLedgerPack purchaseLedgerPack)
         {
             this.exchangeRateRepository = exchangeRateRepository;
             this.ledgerPeriodRepository = ledgerPeriodRepository;
+            this.supplierRepository = supplierRepository;
+            this.orderDetailRepository = orderDetailRepository;
+            this.transactionManager = transactionManager;
+            this.purchaseLedgerPack = purchaseLedgerPack;
+            this.purchaseLedgerRepository = purchaseLedgerRepository;
+        }
+
+        public IEnumerable<ImportBookExchangeRate> GetExchangeRates(string date)
+        {
+            var unformattedDate = DateTime.Parse(date);
+            var formattedDate = unformattedDate.ToString("MMMYYYY");
+
+            var ledgerPeriod = this.ledgerPeriodRepository.FindBy(x => x.MonthName == formattedDate);
+
+            var exchangeRates = this.exchangeRateRepository.FilterBy(x => x.PeriodNumber == ledgerPeriod.PeriodNumber);
+
+            return exchangeRates;
+        }
+
+        public ProcessResult PostDutyForOrderDetails(
+            IEnumerable<ImportBookOrderDetail> orderDetails,
+            int supplierId,
+            int employeeId,
+            DateTime postDutyDate)
+        {
+            foreach (var detail in orderDetails)
+            {
+                var oldDetail = this.orderDetailRepository.FindById(
+                    new ImportBookOrderDetailKey(detail.ImportBookId, detail.LineNumber));
+                if (oldDetail == null)
+                {
+                    throw new PostDutyException("Cannot post duty without saving all order details first");
+                }
+
+                if (string.IsNullOrEmpty(oldDetail.PostDuty) && !string.IsNullOrEmpty(detail.PostDuty) && detail.PostDuty.Equals("Y"))
+                {
+                    this.PostDuty(detail, supplierId, employeeId, postDutyDate);
+
+                    oldDetail.PostDuty = "Y";
+                    this.transactionManager.Commit();
```

Comment:
i don't like transactions being committed in the domain (although i did do it once myself and didn't enjoy it and really it should refactor it out). 
If you've got a dedicated facade service method for this anyway might be better to commit it there?

Reply:
> the first time Rhona uses this it would be a good idea for you or her to let Leigh know to carefully check the resulting purchase ledger post.

Agreed, will do. I'll wait until Jenkins is back before merging